### DISCUSSION TITLE
$process-message operation

### DISCUSSION THREAD

**Discussion by LodewijkSioen** [FirelyTeam/firely-net-sdk#2601] [Ideas]
We're working with [FHIR messaging](http://hl7.org/fhir/R4/messaging.html). For this we opted to use the [$process-message operation](http://hl7.org/fhir/R4/messageheader-operation-process-message.html). This line in the specifications is giving me headaches:
```
This operation does not use the parameters resource; the parameters "async" and "response-url" always go in the URL, if they are used, and the "content" parameter is always the body of the HTTP message.
```
I don't think it is possible to do this with the FhirClient. All the `operation`-based functions take a `Parameter` object and post it to the server.

Is this something that is missing on the SDK?

**Comment by ewoutkramer** [comment]
Yeah, it seems we need to add explicit support for message-based operations where the body is simply the message and does not use parameters at all. I'll convert this to an issue.

**Reply by LodewijkSioen** [reply]
Hold your horses, I already made this in an issue and sent a PR ;)
